### ISSUE TITLE
Geometry#isValid() throws NullPointerException

### ISSUE THREAD

**Issue by sloob** [locationtech/jts#356]
After editing a geometry we get a result geometry like „MULTIPOLYGON (((30 10, 40 40, 20 40, 10 20, 30 10)), EMPTY)“.
With this geometry, the isValid() method throws a NullPointerException (master-branch). With version 1.14.0 (which we actually use) we get an IndexOutOfBoundsException.
To reproduce the issue you can use the following code:

```
GeometryFactory geometryFactory = new GeometryFactory();
        WKTReader reader = new WKTReader(geometryFactory);
        try {
            MultiPolygon point = (MultiPolygon) reader.read("MULTIPOLYGON (((30 10, 40 40, 20 40, 10 20, 30 10)), EMPTY)");
            System.out.println("valid: " + point.isValid());
        } catch (Exception exc) {
        	System.out.println(exc);
        }
}
```

Is the specified geometry not valid?
Nevertheless, the method should return false if the geometry is not valid.

As a workaround, we remove the EMPTY parts, so the exception does not occure.

**Comment by dr-jts** [comment]
This is a bug.  I believe the geometry is valid (although not very useful), but as you say the isValid method should not throw an exception.  Luckily, this should be an easy fix.

**Comment by dr-jts** [comment]
Also, having EMPTY components of MultiGeometries is not a good thing.  What is causing the EMPTY component to appear?  Is is a JTS operation?  Or if not can you fix the upstream code?

**Comment by dr-jts** [comment]
GEOS already has a fix for this:  https://github.com/libgeos/geos/commit/f2d7001ae817d151ec95daa8451834212b3301d2

**Comment by dr-jts** [comment]
Closed by #360
